src/plugins/es_ui_shared/public/console_lang/ace/modes/x_json/x_json.ts

```diff
@@ -18,23 +18,50 @@
  */
 
 import ace from 'brace';
-
 import { XJsonHighlightRules } from '../index';
+import { workerModule } from './worker';
+
+const { WorkerClient } = ace.acequire('ace/worker/worker_client');
 
 const oop = ace.acequire('ace/lib/oop');
+
 const { Mode: JSONMode } = ace.acequire('ace/mode/json');
 const { Tokenizer: AceTokenizer } = ace.acequire('ace/tokenizer');
 const { MatchingBraceOutdent } = ace.acequire('ace/mode/matching_brace_outdent');
 const { CstyleBehaviour } = ace.acequire('ace/mode/behaviour/cstyle');
 const { FoldMode: CStyleFoldMode } = ace.acequire('ace/mode/folding/cstyle');
 
-export function XJsonMode(this: any) {
-  const ruleset: any = new XJsonHighlightRules();
+const XJsonMode: any = function XJsonMode(this: any) {
```

Comment:
`this` seems potentially confusing. I know we're not trying to make a lot of out-of-scope changes in this PR, but what do you think of renaming `this` to `target` (or anything else) while we're here?

Reply:
Also, is `const XJsonMode` necessary? Would this declaration work?

```ts
function XJsonMode(this: any): any {
  // ...
}
```

Reply:
Thanks for look through @cjcenizal !

> this seems potentially confusing

The `this: any` annotation is targeting the this value of that function ([docs](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters)). The alternative would be to convert that into a `class` (which shouldn't make a difference to the actual code) but I did not want to change the legacy function usage in this pass.

> Also, is const XJsonMode necessary?
I opted for the non-invasive `const x: any =` to get rid of @ts-ignore statements, otherwise we would need to annotate the `new` value to satisfy TS. I figured since it will be `any` anyway I just added the annotation over the whole function.

Reply:
Thanks for the explanation JL! TIL.